File: src/integration-tests/expecteds/py/json-schema-meta-schema.py
from typing import NewType

$Id = NewType("$Id", str)
from typing import NewType

$Schema = NewType("$Schema", str)
from typing import NewType

$Ref = NewType("$Ref", str)
from typing import NewType

$Comment = NewType("$Comment", str)
from typing import NewType

Title = NewType("Title", str)
from typing import NewType

Description = NewType("Description", str)
from typing import Any, NewType

AlwaysTrue = NewType("AlwaysTrue", Any)
from typing import NewType

ReadOnly = NewType("ReadOnly", bool)
from typing import List, NewType

Examples = NewType("Examples", List[AlwaysTrue])
from typing import NewType

MultipleOf = NewType("MultipleOf", float)
from typing import NewType

Maximum = NewType("Maximum", float)
from typing import NewType

ExclusiveMaximum = NewType("ExclusiveMaximum", float)
from typing import NewType

Minimum = NewType("Minimum", float)
from typing import NewType

ExclusiveMinimum = NewType("ExclusiveMinimum", float)
from typing import NewType

NonNegativeInteger = NewType("NonNegativeInteger", int)
from typing import Any, NewType

DefaultZero = NewType("DefaultZero", Any)
from typing import NewType, Any, Mapping

NonNegativeIntegerDefaultZero = NewType("NonNegativeIntegerDefaultZero", Mapping[Any, Any])
from typing import NewType

Pattern = NewType("Pattern", str)
from typing import List, NewType

SchemaArray = NewType("SchemaArray", List[JSONMetaSchema])
from typing import NewType, Union

Items = NewType("Items", Union[JSONMetaSchema, SchemaArray])
from typing import NewType

UniqueItems = NewType("UniqueItems", bool)
from typing import NewType

StringDoaGddGA = NewType("StringDoaGddGA", str)
from typing import List, NewType

StringArray = NewType("StringArray", List[StringDoaGddGA])
from typing import NewType, Any, Mapping

Definitions = NewType("Definitions", Mapping[Any, Any])
from typing import NewType, Any, Mapping

Properties = NewType("Properties", Mapping[Any, Any])
from typing import NewType, Any, Mapping

PatternProperties = NewType("PatternProperties", Mapping[Any, Any])
from typing import NewType, Union

AnyOfJSONMetaSchemaStringArrayBk9CjcAm = NewType("AnyOfJSONMetaSchemaStringArrayBk9CjcAm", Union[JSONMetaSchema, StringArray])
from typing import NewType, Any, Mapping

Dependencies = NewType("Dependencies", Mapping[Any, Any])
from typing import List, NewType

Enum = NewType("Enum", List[AlwaysTrue])
from typing import Any, NewType

SimpleTypes = NewType("SimpleTypes", Any)
from typing import List, NewType

ArrayOfSimpleTypes = NewType("ArrayOfSimpleTypes", List[SimpleTypes])
from typing import NewType, Union

AnyOfArrayOfSimpleTypesSimpleTypesBOu2T0PK = NewType("AnyOfArrayOfSimpleTypesSimpleTypesBOu2T0PK", Union[SimpleTypes, ArrayOfSimpleTypes])
from typing import NewType

Format = NewType("Format", str)
from typing import NewType

ContentMediaType = NewType("ContentMediaType", str)
from typing import NewType

ContentEncoding = NewType("ContentEncoding", str)
from typing import TypedDict, Optional

class JSONMetaSchema(TypedDict):
    $id: Optional[$Id]
    $schema: Optional[$Schema]
    $ref: Optional[$Ref]
    $comment: Optional[$Comment]
    title: Optional[Title]
    description: Optional[Description]
    default: Optional[AlwaysTrue]
    readOnly: Optional[ReadOnly]
    examples: Optional[Examples]
    multipleOf: Optional[MultipleOf]
    maximum: Optional[Maximum]
    exclusiveMaximum: Optional[ExclusiveMaximum]
    minimum: Optional[Minimum]
    exclusiveMinimum: Optional[ExclusiveMinimum]
    maxLength: Optional[NonNegativeInteger]
    minLength: Optional[NonNegativeIntegerDefaultZero]
    pattern: Optional[Pattern]
    additionalItems: Optional[JSONMetaSchema]
    items: Optional[Items]
    maxItems: Optional[NonNegativeInteger]
    minItems: Optional[NonNegativeIntegerDefaultZero]
    uniqueItems: Optional[UniqueItems]
    contains: Optional[JSONMetaSchema]
    maxProperties: Optional[NonNegativeInteger]
    minProperties: Optional[NonNegativeIntegerDefaultZero]
    required: Optional[StringArray]
    additionalProperties: Optional[JSONMetaSchema]
    definitions: Optional[Definitions]
    properties: Optional[Properties]
    patternProperties: Optional[PatternProperties]
    dependencies: Optional[Dependencies]
    propertyNames: Optional[JSONMetaSchema]
    const: Optional[AlwaysTrue]
    enum: Optional[Enum]
    type: Optional[AnyOfArrayOfSimpleTypesSimpleTypesBOu2T0PK]
    format: Optional[Format]
    contentMediaType: Optional[ContentMediaType]
    contentEncoding: Optional[ContentEncoding]
    if: Optional[JSONMetaSchema]
    then: Optional[JSONMetaSchema]
    else: Optional[JSONMetaSchema]
    allOf: Optional[SchemaArray]
    anyOf: Optional[SchemaArray]
    oneOf: Optional[SchemaArray]
    not: Optional[JSONMetaSchema]

File: src/integration-tests/expecteds/ts/json-schema-meta-schema.ts
export type $Id = string;
export type $Schema = string;
export type $Ref = string;
export type $Comment = string;
export type Title = string;
export type Description = string;
type AlwaysTrue = any;
export type ReadOnly = boolean;
export type Examples = AlwaysTrue[];
export type MultipleOf = number;
export type Maximum = number;
export type ExclusiveMaximum = number;
export type Minimum = number;
export type ExclusiveMinimum = number;
export type NonNegativeInteger = number;
export type DefaultZero = any;
export type NonNegativeIntegerDefaultZero = NonNegativeInteger & DefaultZero;
export type Pattern = string;
export type SchemaArray = JSONMetaSchema[];
/**
 *
 * @default true
 *
 */
export type Items = JSONMetaSchema | SchemaArray;
export type UniqueItems = boolean;
export type StringDoaGddGA = string;
/**
 *
 * @default []
 *
 */
export type StringArray = StringDoaGddGA[];
/**
 *
 * @default {}
 *
 */
export interface Definitions { [key: string]: any; }
/**
 *
 * @default {}
 *
 */
export interface Properties { [key: string]: any; }
/**
 *
 * @default {}
 *
 */
export interface PatternProperties { [key: string]: any; }
export type AnyOfJSONMetaSchemaStringArrayBk9CjcAm = JSONMetaSchema | StringArray;
export interface Dependencies { [key: string]: any; }
export type Enum = AlwaysTrue[];
export type SimpleTypes = any;
export type ArrayOfSimpleTypes = SimpleTypes[];
export type AnyOfArrayOfSimpleTypesSimpleTypesBOu2T0PK = SimpleTypes | ArrayOfSimpleTypes;
export type Format = string;
export type ContentMediaType = string;
export type ContentEncoding = string;
/**
 *
 * @default true
 *
 */
export interface JSONMetaSchema {
  $id?: $Id;
  $schema?: $Schema;
  $ref?: $Ref;
  $comment?: $Comment;
  title?: Title;
  description?: Description;
  default?: AlwaysTrue;
  readOnly?: ReadOnly;
  examples?: Examples;
  multipleOf?: MultipleOf;
  maximum?: Maximum;
  exclusiveMaximum?: ExclusiveMaximum;
  minimum?: Minimum;
  exclusiveMinimum?: ExclusiveMinimum;
  maxLength?: NonNegativeInteger;
  minLength?: NonNegativeIntegerDefaultZero;
  pattern?: Pattern;
  additionalItems?: JSONMetaSchema;
  items?: Items;
  maxItems?: NonNegativeInteger;
  minItems?: NonNegativeIntegerDefaultZero;
  uniqueItems?: UniqueItems;
  contains?: JSONMetaSchema;
  maxProperties?: NonNegativeInteger;
  minProperties?: NonNegativeIntegerDefaultZero;
  required?: StringArray;
  additionalProperties?: JSONMetaSchema;
  definitions?: Definitions;
  properties?: Properties;
  patternProperties?: PatternProperties;
  dependencies?: Dependencies;
  propertyNames?: JSONMetaSchema;
  const?: AlwaysTrue;
  enum?: Enum;
  type?: AnyOfArrayOfSimpleTypesSimpleTypesBOu2T0PK;
  format?: Format;
  contentMediaType?: ContentMediaType;
  contentEncoding?: ContentEncoding;
  if?: JSONMetaSchema;
  then?: JSONMetaSchema;
  else?: JSONMetaSchema;
  allOf?: SchemaArray;
  anyOf?: SchemaArray;
  oneOf?: SchemaArray;
  not?: JSONMetaSchema;
  [k: string]: any;
}
